File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let curIndex =0;


    //TODO 0이 아닐때
    for(let idx = 0 ; idx < nums.length ; idx++){
        if(nums[idx] != 0){
            nums[curIndex] = nums[idx];
            curIndex++;
        }
    }

    //0이 아닌거 찾은 이후에 넣기 
    for(let i = curIndex ; i < nums.length ;i++){
        nums[i] = 0; 
    }

    return nums
};

File: 0283-move-zeroes/0283-move-zeroes.go
func moveZeroes(nums []int)  {
    // 0을 끝으로 옮김moveZeroes

    startIndex := 0 

    for i := 0 ; i < len(nums) ; i++ {
        if nums[i] != 0 {
            nums[startIndex] = nums[i]
            startIndex++;
        }
    }

    for startIndex < len(nums) {
        nums[startIndex] = 0;
        startIndex++;
    }

}